setup.py

```diff
@@ -30,7 +30,7 @@
         'ecmwf-api-client',
         'eofs',
         'ESMPy',
-        'esmvalcore',
+        'esmvalcore>=2.7',
```

Comment:
There should be an upper pin too (as we previously had: https://github.com/ESMValGroup/ESMValTool/blob/2190646b6e5fbe49a1c00f1f209891b1883f7a19/setup.py#L36) because we test ESMValTool with a specific version of ESMValCore we pin to that version. Is `2.8.0-dev` smaller than `2.8`? Then we could just do `esmvalcore >=2.7,<2.8`.

Reply:
good point and good question, I'll investigate about `dev` being a wee bit smaller

Reply:
OK I tested the pin for the pip installation too - which goes through but it complains (it's an error but it's not raised to the system) `esmvaltool 2.8.0.dev46+g2a3a15f1c requires esmvalcore<2.8,>=2.7, but you have esmvalcore 2.8.0.dev56+gd6373e75d which is incompatible.` (see the GA test) - so pip doesn't care much for semantic versioning, which is a bit of a bummer. Not a make or break deal though

Reply:
What if you try `esmvalcore >=2.7,<=2.8.0.dev`?

Reply:
I'm not sure if my comment above is quite right: actually our promise is that ESMValTool v2.8.0-dev is compatible with ESMValCore v2.7.* and then we fix any incompatibility issues in the period between making the ESMValCore v2.8.0 release and making the ESMValTool v2.8.0 release.

Reply:
Did you try if you can run the `esmvaltool` command if you install the wrong version of ESMValCore?

Reply:
yeah that should work fine - the GA test is calling `esmvaltool` after it pip installs it. I think `<=2.8dev` may be OK only in setup.py

Reply:
wow pip is really dumb: `esmvaltool 2.8.0.dev47+g04088cdaf requires esmvalcore<=2.8.dev,>=2.7, but you have esmvalcore 2.8.0.dev57+g28379023a which is incompatible.` How about we just not upper pin in setup.py?

Reply:
That would pull in the wrong version of ESMValCore when people install older versions of ESMValTool.

Reply:
huh? it'll still be limited by the lower pin, >=2.7. You know what? I may just dump this PR entirely, if we get hit by a situation when the pin is affecting the installation we'll just have to instruct the user how to set it manually. What you reckon @bouweandela ?

Reply:
What about trying `v2.8.0` as an upper pin? That should still allow `dev` and `rc` versions, but not the actual next release.

Reply:
So how about 2.9.0 as we have just discussed in person 1 minute ago? 😁